src/Core/AdminConsole/Utilities/DebuggingInstruments/UserInviteDebuggingLogger.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Bit.Core.Entities;
+using Bit.Core.Enums;
+using Microsoft.Extensions.Logging;
+using Quartz.Util;
+
+namespace Bit.Core.AdminConsole.Utilities.DebuggingInstruments;
+
+public class UserInviteDebuggingLogger(ILogger<UserInviteDebuggingLogger> logger) : IUserInviteDebuggingLogger
+{
+    public void Log(OrganizationUser allOrgUser)
+    {
+        Log([allOrgUser]);
+    }
+
+    public void Log(IEnumerable<OrganizationUser> allOrgUsers)
+    {
+        try
+        {
+            var invalidInviteState = allOrgUsers.Any(user => user.Status == OrganizationUserStatusType.Invited && user.Email.IsNullOrWhiteSpace());
+
+            if (invalidInviteState)
+            {
+                logger.LogWarning("Warning invalid invited state.");
+            }
+
+            var invalidNoneInviteState = allOrgUsers.Any(user => user.Status != OrganizationUserStatusType.Invited && !user.Email.IsNullOrWhiteSpace());
+
+            if (invalidNoneInviteState)
+            {
+                logger.LogWarning("Warning invalid non invited state.");
+            }
```

Comment:
This will throw a false positive for users who have been invited and then revoked; their status will be revoked (`!= Invited`) but their email will not be null. I suggest being explicit and limiting it to `Accepted` and `Confirmed` states. I think we can probably ignore revoked here.

(For this reason, I think that `Revoked` should be a separate flag rather than an `OrganizationUserStatusType`, because today it hides their "real" or underlying status, and we have kinda [hacky logic](https://github.com/bitwarden/server/blob/4a1212095053bb93e936038a71f44b01ebbb3c36/src/Core/AdminConsole/Services/Implementations/OrganizationService.cs#L1691-L1708) to figure out what to restore them to.)